File: frontend/src/types/conversation.ts
export interface Conversation {
  id: number;
  user_id: number;
  title: string | null;
  model: string | null;
  temperature: number | null;
  max_tokens: number | null;
  created_at: string;
  updated_at: string | null;
  messages?: Message[];
}

export interface ConversationListItem {
  id: number;
  user_id: number;
  title: string | null;
  model: string | null;
  created_at: string;
  updated_at: string | null;
  message_count: number;
}

export interface Message {
  id: number;
  conversation_id: number;
  role: 'user' | 'assistant' | 'system';
  content: string;
  usage?: {
    prompt_tokens?: number;
    completion_tokens?: number;
    total_tokens?: number;
  };
  created_at: string;
}

export interface ConversationCreate {
  title?: string;
  model?: string;
  temperature?: number;
  max_tokens?: number;
}

File: backend/app/models/conversation.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey, DateTime, Text, JSON
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base


class Conversation(Base):
    """대화 세션 모델"""
    __tablename__ = "conversations"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    title = Column(String, nullable=True)  # 대화 제목 (첫 메시지 기반)
    model = Column(String, nullable=True)  # 사용된 모델
    temperature = Column(Float, nullable=True)  # 사용된 temperature
    max_tokens = Column(Integer, nullable=True)  # 사용된 max_tokens
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # 관계
    messages = relationship("Message", back_populates="conversation", cascade="all, delete-orphan")


class Message(Base):
    """대화 메시지 모델"""
    __tablename__ = "messages"

    id = Column(Integer, primary_key=True, index=True)
    conversation_id = Column(Integer, ForeignKey("conversations.id"), nullable=False, index=True)
    role = Column(String, nullable=False)  # user, assistant, system
    content = Column(Text, nullable=False)  # 메시지 내용
    usage = Column(JSON, nullable=True)  # 토큰 사용량 정보
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # 관계
    conversation = relationship("Conversation", back_populates="messages")

